src/cmd/windows/durable_task_monitor.go

```diff
@@ -0,0 +1,179 @@
+			return
+		}
+		defer outputFile.Close()
+
+		// Note: Go writes the output in utf8 WITHOUT a bom. No need for any encoding conversions
+		scriptCmd.Stdout = outputFile
+		scriptCmd.Stderr = scriptLogger.Writer()
+	} else {
+		scriptCmd.Stdout = scriptLogger.Writer()
+		scriptCmd.Stderr = scriptCmd.Stdout
+	}
+
+	for i := 0; i < len(scriptCmd.Args); i++ {
+		launchLogger.Printf("args %v: %v\n", i, scriptCmd.Args[i])
+	}
+	err := scriptCmd.Start()
+	if common.CheckIfErr(scriptLogger, err) {
+		common.ExitLauncher(-2, resultPath, scriptLogger)
+		return
+	}
+	pid := scriptCmd.Process.Pid
+	launchLogger.Printf("launched %v\n", pid)
+	err = scriptCmd.Wait()
+	common.CheckIfErr(scriptLogger, err)
+	resultVal := scriptCmd.ProcessState.ExitCode()
+	launchLogger.Printf("script exit code: %v\n", resultVal)
+
+	common.ExitLauncher(resultVal, resultPath, launchLogger)
+}
+
+func main() {
+	var controlDir, resultPath, logPath, shell, scriptPath, outputPath string
+	var debug, daemon bool
+	const controlFlag = "controldir"
+	const resultFlag = "result"
+	const logFlag = "log"
+	const shellFlag = "shell"
+	const scriptPathFlag = "script"
+	const outputFlag = "output"
+	const debugFlag = "debug"
+	const daemonFlag = "daemon"
+	flag.StringVar(&controlDir, controlFlag, "", "working directory")
+	flag.StringVar(&resultPath, resultFlag, "", "full path of the result file")
+	flag.StringVar(&logPath, logFlag, "", "full path of the log file")
+	flag.StringVar(&scriptPath, scriptPathFlag, "", "full path of the script to be launched")
+	flag.StringVar(&shell, shellFlag, "cmd", "Windows shell type")
+	flag.StringVar(&outputPath, outputFlag, "", "(optional) if recording output, full path of the output file")
+	flag.BoolVar(&debug, debugFlag, false, "noisy output to log")
+	flag.BoolVar(&daemon, daemonFlag, false, "Free binary from parent process")
+	flag.Parse()
+
+	// Validate that the required flags were all command-line defined
+	// required := []string{scriptPathFlag}
+	required := []string{controlFlag, resultFlag, logFlag, scriptPathFlag}
+	defined := make(map[string]string)
+	flag.Visit(func(f *flag.Flag) {
+		defined[f.Name] = f.Value.String()
+	})
+	common.ValidateFlags(defined, required)
+
+	fmt.Fprintf(os.Stdout, "Parent pid is: %v\n", os.Getppid())
+
+	if daemon {
+		fmt.Fprintf(os.Stdout, "1st launch pid is: %v\n", os.Getpid())
+		rebuiltArgs := common.RebuildArgs(defined, daemonFlag)
+		doubleLaunchCmd := exec.Command(os.Args[0], rebuiltArgs...)
+		doubleLaunchCmd.Stdout = nil
+		doubleLaunchCmd.Stderr = nil
+		doubleLaunchCmd.SysProcAttr = &syscall.SysProcAttr{CreationFlags: windows.DETACHED_PROCESS | windows.CREATE_NEW_PROCESS_GROUP}
+		doubleLaunchErr := doubleLaunchCmd.Start()
+		if doubleLaunchErr != nil {
+			panic("Double launch failed, exiting")
+		}
+		return
+	}
+	// Prepare logging
+	logFile, logErr := os.Create(logPath)
+	if logErr != nil {
+		fmt.Fprintf(os.Stderr, "Unable to create log file: %s", logErr)
+		return
```

Comment:
should we not `Exit` with an errorcode here?

Reply:
resolved in 8b29388f033068b9a614df700aaea11881d72f05